### PR TITLE
Force xcb platform only on Gnome

### LINKED ISSUE #170: Enabling wayland back
We are forcing QT_QPA_PLATFORM=xcb right now. We don't need to live like this forever.

I don't know what needs to happen for it to get back to native, is it documented somewhere?

### PR DESCRIPTION
**Pull request by gasinvein**
Telegram should work fine now on other wayland DEs. Testing wanted.
Partially fixes #170.

### CODE REVIEW

**Comment by gasinvein** [comment]
Hm, I guess this won't work for autostarted Telegram because it still creates .desktop entry with `--command=telegram-desktop` in `Exec`.

**Comment by aleixpol** [comment]
This is unfortunate but if it's what it takes, it's better than what we have today.

File: telegram-wrapper.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+if [ "${XDG_CURRENT_DESKTOP,,}" == "gnome" ] || [ "${XDG_SESSION_DESKTOP,,}" == "gnome" ]; then
```
Comment (SnejUgal) [inline telegram-wrapper.sh:3]: This strict check ignores non-vanilla Gnome sessions, e.g. on Ubuntu these variables are `ubuntu:GNOME` and `ubuntu-wayland` respectively, Manjaro's Gnome seems to change them too.

**Comment by grulja** [comment]
+1 from me, as far as I know this was broken only on GNOME. As mentioned in the comment above, we should maybe compare if XDG_CURRENT_DESKTOP starts with "GNOME".

File: telegram-wrapper.sh (hunk 1 above)
Comment (gasinvein) [reply telegram-wrapper.sh:3]: Can you please specify which of this vars (`XDG_CURRENT_DESKTOP` and `XDG_SESSION_DESKTOP`) contains which value on which session?

**Comment by ilya-fedin** [comment]
+1 from mr as well

File: telegram-wrapper.sh (hunk 1 above)
Comment (SnejUgal) [reply telegram-wrapper.sh]: Session | `XDG_CURRENT_DESKTOP` | `XDG_SESSION_DESKTOP`
---|---|---
Ubuntu | `ubuntu:GNOME` | `ubuntu-wayland`
Ubuntu with `communitheme` | `communitheme:ubuntu:GNOME` | `ubuntu-communitheme-snap-wayland`
Manjaro | `GNOME` | `gnome`
Manjaro with Yaru AUR | `Yaru:ubuntu:GNOME` | `Yaru-wayland`

(`communitheme` is Yaru backported to 18.04; for some reason Yaru requires its own session)

Comment (gasinvein) [reply telegram-wrapper.sh]: Thanks. Matching `gnome` substring should work, then.

**Pull request merged**